src/VCR/Cassette.php

```diff
@@ -68,8 +69,24 @@ public function hasResponse(Request $request)
      */
     public function playback(Request $request)
     {
+        return $this->getResponse($request);
+    }
+
+    /**
+     * Returns a response for given request or null if not found.
+     * Optionally dispatches VCR_BEFORE_PLAYBACK event.
+     *
+     * @param Request $request Request to check if it was recorded.
+     * @param boolean $dispatchEvents Whether to dispatch events or not. Default is false.
+     *
+     * @return boolean True if a response was recorded for specified request.
+     */
+    protected function getResponse(Request $request, $dispatchEvents = false) {
         foreach ($this->storage as $recording) {
             $storedRequest = Request::fromArray($recording['request']);
+            if ($dispatchEvents) {
+                $this->config->dispatch(VCREvents::VCR_BEFORE_PLAYBACK, new BeforePlaybackEvent($request, $this));
```

Comment:
I think that the request of the BeforePlaybackEvent should be the storedRequest.